ihmc-high-level-behaviors/src/main/java/us/ihmc/perception/StandAloneRealsenseProcess.java

```diff
@@ -0,0 +1,89 @@
+
+import java.util.function.Supplier;
+
+/**
+ * This class handles publishing the color and depth of the realsense. Its meant to be a stand alone class that only touches the realsense.
+ */
+public class StandAloneRealsenseProcess
+{
+   private static final String REALSENSE_SERIAL_NUMBER = System.getProperty("d455.serial.number", "213522252883");
+   private static final ROS2Topic<ImageMessage> REALSENSE_COLOR_TOPIC = PerceptionAPI.D455_COLOR_IMAGE;
+   private static final ROS2Topic<ImageMessage> REALSENSE_DEPTH_TOPIC = PerceptionAPI.D455_DEPTH_IMAGE;
+
+   private final RealsenseColorDepthImageRetriever realsenseImageRetriever;
+   private final RealsenseColorDepthImagePublisher realsenseImagePublisher;
+
+   private final ROS2DemandGraphNode realsenseDemandNode;
+   private Supplier<ReferenceFrame> realsenseFrameSupplier = ReferenceFrame::getWorldFrame;
+
+   public StandAloneRealsenseProcess(ROS2Helper ros2Helper, ROS2SyncedRobotModel syncedRobot)
+   {
+      if (syncedRobot != null)
+      {
+         realsenseFrameSupplier = syncedRobot.getReferenceFrames()::getSteppingCameraFrame;
+      }
+      realsenseDemandNode = new ROS2DemandGraphNode(ros2Helper, PerceptionAPI.REQUEST_REALSENSE_POINT_CLOUD);
+
+      realsenseImageRetriever = new RealsenseColorDepthImageRetriever(new RealsenseDeviceManager(),
+                                                                      REALSENSE_SERIAL_NUMBER,
+                                                                      RealsenseConfiguration.D455_COLOR_720P_DEPTH_720P_30HZ,
+                                                                      realsenseFrameSupplier,
+                                                                      realsenseDemandNode);
+
+      realsenseImagePublisher = new RealsenseColorDepthImagePublisher(REALSENSE_DEPTH_TOPIC, REALSENSE_COLOR_TOPIC);
+
+      RestartableThread realsenseProcessAndPublishThread = new RestartableThread("RealsenseProcess", this::publishRealSense);
+      realsenseProcessAndPublishThread.start();
+   }
+
+   private void publishRealSense()
+   {
+      if (realsenseDemandNode.isDemanded())
+      {
+         RawImage realsenseDepthImage = realsenseImageRetriever.getLatestRawDepthImage();
+         RawImage realsenseColorImage = realsenseImageRetriever.getLatestRawColorImage();
+
+         realsenseImagePublisher.setNextDepthImage(realsenseDepthImage.get());
+         realsenseImagePublisher.setNextColorImage(realsenseColorImage.get());
+
+         realsenseDepthImage.release();
+         realsenseColorImage.release();
+      }
+      else
+         ThreadTools.sleep(500);
+   }
+
+   public void destroy()
+   {
+      realsenseImagePublisher.destroy();
+      realsenseImageRetriever.destroy();
+      realsenseDemandNode.destroy();
+   }
+
+   public static void main(String[] args)
+   {
+      ROS2Node ros2Node = ROS2Tools.createROS2Node(DomainFactory.PubSubImplementation.FAST_RTPS, "nadia_realsense_process");
```

Comment:
This ROS2Node should be destroyed too, right?

Reply:
Nice yep got it